frontend/scenes/GameScene.js

```diff
@@ -18,58 +22,81 @@ export default class GameScene extends Phaser.Scene {
+  }
+
   create() {
     this.add.image(0, 0, "bg").setOrigin(0, 0).setScale(0.5);
-    this.cardMechanics.createCard( // first middle card
-        gameRules.pilePosition.x,
-        gameRules.pilePosition.y,
-        // imageComb
-        [0, 1, 2, 3, 4, 5, 6, 7], // FIXME send from backend instead
-        "pile",
-        0xfff00,
-        () => {}
-    );
-    this.setPlayersCard();
     this.createVictorySign();
-    this.events.on("anotherPlayerScored", this.setMiddleCard, this);
-    this.events.on("gameEnd", this.gameEnd, this);
-    this.simulateOtherPlayerScoring();
     this.errorMessage = new ErrorMessage(
-        this,
-        this.scale.width / 2,
-        40,
-        this.scale.width * 0.9,
+      this,
+      this.scale.width / 2,
+      40,
+      this.scale.width * 0.9
     );
+    this.ws.onmessage = (ev) => {
+      const message = JSON.parse(ev.data);
+      if (message.method === "init") {
+        this.initializeGame(message);
+      } else if (message.method === "score") {
+        this.handleScore(message);
+      } else if (message.method === "end") {
+        this.gameEnd(message.winner);
+      }
+    };
+    if (this.isHost) {
+      this.ws.send(
+        JSON.stringify(this.buildInitMessage(playerInfo.id, gameState.gameId))
+      );
+    }
   }
 
-  setPlayersCard() {
-    // FIXME fetch imageCombination from GameState
-    this.cardMechanics.updatePlayersCard([0, 1, 2, 3, 4, 5, 6, 7], () => {
-      this.setPlayersCard();
-    });
+  initializeGame(initMessage) {
+    gameState.middleCard = initMessage.middle_card;
+    this.cards = initMessage.cards;
+    this.cardMechanics.createCard(
+      gameRules.pilePosition.x,
+      gameRules.pilePosition.y,
+      initMessage.middle_card,
+      "pile",
+      0x00000,
+      () => {}
+    );
+    this.setPlayersCard(this.cards.pop());
   }
 
-  setMiddleCard() {
-    // FIXME fetch imageCombination from GameState
-    const otherPlayersCard = this.cardMechanics.updateMiddleCard(
-            [0, 1, 2, 3, 4, 5, 6, 7], "violette", 0x7f00ff, () => {}
-        );
+  handleScore(scoreMessage) {
+    if (scoreMessage.player_id === playerInfo.id) {
+      this.cardMechanics.score(this.currentCard);
+      this.setPlayersCard(this.cards.pop());
+    } else {
+      gameState.blocked = false;
+      this.otherPlayerScore(
+        scoreMessage.new_middle_card,
+        otherPlayers.find((player) => player.id === scoreMessage.player_id)
+      );
+    }
   }
 
-  simulateOtherPlayerScoring() {
-    this.time.addEvent({
-      delay: 2000,
-      callback: () => {
-        this.events.emit("anotherPlayerScored");
-      },
-      loop: this.isGameActive(),
-    });
+  setPlayersCard(playersCard) {
+    this.currentCard = this.cardMechanics.updatePlayersCard(playersCard);
+  }
+
+  otherPlayerScore(newMiddleCard, player) {
+    this.cardMechanics.updateMiddleCardWithNewlyCreatedCard(
+      newMiddleCard,
+      player
+    );
   }
 
   gameEnd(winner) {
```

Comment:
todo: close ws connection from client side here (backend does it already but just in case)